openshift/patches/015-srvke-456.patch

```diff
@@ -0,0 +1,350 @@
+diff --git a/pkg/apis/sources/v1beta1/sinkbinding_context.go b/pkg/apis/sources/v1beta1/sinkbinding_context.go
+index 787f91ca3..8f4f7672c 100644
+--- a/pkg/apis/sources/v1beta1/sinkbinding_context.go
++++ b/pkg/apis/sources/v1beta1/sinkbinding_context.go
+@@ -20,11 +20,13 @@ import (
+ 	"context"
+ 
+ 	"knative.dev/pkg/apis"
++	"knative.dev/pkg/resolver"
+ )
+ 
+ // sinkURIKey is used as the key for associating information
+ // with a context.Context.
+ type sinkURIKey struct{}
++type resolverKey struct{}
+ 
+ // WithSinkURI notes on the context for binding that the resolved SinkURI
+ // is the provided apis.URL.
+@@ -32,6 +34,10 @@ func WithSinkURI(ctx context.Context, uri *apis.URL) context.Context {
+ 	return context.WithValue(ctx, sinkURIKey{}, uri)
+ }
+ 
++func WithURIResolver(ctx context.Context, resolver *resolver.URIResolver) context.Context {
++	return context.WithValue(ctx, resolverKey{}, resolver)
++}
++
+ // GetSinkURI accesses the apis.URL for the Sink URI that has been associated
+ // with this context.
+ func GetSinkURI(ctx context.Context) *apis.URL {
+@@ -41,3 +47,11 @@ func GetSinkURI(ctx context.Context) *apis.URL {
+ 	}
+ 	return value.(*apis.URL)
+ }
++
++func GetURIResolver(ctx context.Context) *resolver.URIResolver {
++	value := ctx.Value(resolverKey{})
++	if value == nil {
++		return nil
++	}
++	return value.(*resolver.URIResolver)
++}
+diff --git a/pkg/apis/sources/v1beta1/sinkbinding_context_test.go b/pkg/apis/sources/v1beta1/sinkbinding_context_test.go
+index 854cac885..7cbac4095 100644
+--- a/pkg/apis/sources/v1beta1/sinkbinding_context_test.go
++++ b/pkg/apis/sources/v1beta1/sinkbinding_context_test.go
+@@ -41,3 +41,12 @@ func TestGetSinkURI(t *testing.T) {
+ 		t.Errorf("GetSinkURI() = %v, wanted %v", got, want)
+ 	}
+ }
++
++func TestGetURIResolver(t *testing.T) {
++	ctx := context.Background()
++
++	if resolver := GetURIResolver(ctx); resolver != nil {
++		t.Errorf("GetURIResolver() = %v, wanted nil", resolver)
++	}
++
++}
+diff --git a/pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go b/pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go
+index 56de4d8f0..dce06fad1 100644
+--- a/pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go
++++ b/pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go
+@@ -21,6 +21,7 @@ import (
+ 	"encoding/json"
+ 	"fmt"
+ 
++	"go.uber.org/zap"
+ 	corev1 "k8s.io/api/core/v1"
+ 	"k8s.io/apimachinery/pkg/runtime/schema"
+ 	"knative.dev/eventing/pkg/logging"
```

Comment:
seems to not get added to the file. on my box, @lberk 

Let's update this PR, with the actual changes to the code

/hold